src/plugins/flexibleLayout/components/flexibleLayout.vue

```diff
@@ -595,90 +567,52 @@ export default {
         toggleLayoutDirection(v) {
             this.rowsLayout = v;
         },
-        promptBeforeDeletingContainer(containerIndex) {
-            let deleteContainer = this.deleteContainer;
-
-            let prompt = this.openmct.overlays.dialog({
-                iconClass: 'alert',
-                message: `This action will permanently delete container ${containerIndex + 1} from this Flexible Layout`,
-                buttons: [
-                    {
-                        label: 'Ok',
-                        emphasis: 'true',
-                        callback: function () {
-                            deleteContainer(containerIndex);
-                            prompt.dismiss();
-                        },
-                    },
-                    {
-                        label: 'Cancel',
-                        callback: function () {
-                            prompt.dismiss();
-                        }
-                    }
-                ]
-            });
-        },
         deleteContainer(containerIndex) {
             this.containers.splice(containerIndex, 1);
 
             this.recalculateContainerSize(100/this.containers.length);
             this.persist();
         },
-        addContainer(containerIndex) {
-            let newContainer = new Container();
-
-            if (typeof containerIndex === 'number') {
-                this.containers.splice(containerIndex+1, 0, newContainer);
-            } else {
-
-                this.containers.push(newContainer);
-            }
-
-            this.recalculateContainerSize(100/this.containers.length);
+        deleteFrame(frameIndex, containerIndex) {
+            this.containers[containerIndex].frames.splice(frameIndex, 1);
```

Comment:
you don't want to store state in two places (the domain object, and this.containers).
1. you should mutate the domain object directly from here.
2. an observer on the domain object should update containers.

Reply:
I changed the observe callback to update this.domainObject as well. This throws a vue warning, because domainObject is injected into flexibleLayout, but it does fix the bug you mentioned

Reply:
Should follow vuejs guide on [one way data flow](https://vuejs.org/v2/guide/components-props.html#One-Way-Data-Flow) to prevent this error.